core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -83,9 +85,30 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
     val keyOrdering: Option[Ordering[K]] = None,
     val aggregator: Option[Aggregator[K, V, C]] = None,
     val mapSideCombine: Boolean = false,
-    val shuffleWriterProcessor: ShuffleWriteProcessor = new ShuffleWriteProcessor)
+    val shuffleWriterProcessor: ShuffleWriteProcessor = new ShuffleWriteProcessor,
+    val rowBasedChecksums: Array[RowBasedChecksum] = Array.empty)
```

Comment:
`Array.empty` is not a constant but a function. Shall we create a constant of empty `Array[RowBasedChecksum]` in `object ShuffleDependency`?